src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/UnscopedRefAttribute.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    /// Used to indicate a byref escapes and is not scoped.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// There are several cases where the C# compiler treats a <see langword="ref"/> as implicitly
+    /// <see langword="scoped"/> - where the compiler does not allow the <see langword="ref"/> to escape the method.
+    /// </para>
+    /// <para>
+    /// For example:
+    /// <list type="number">
+    ///     <item><see langword="this"/> for <see langword="struct"/> instance methods.</item>
+    ///     <item><see langword="ref"/> parameters that refer to <see langword="ref"/> <see langword="struct"/> types.</item>
+    ///     <item><see langword="out"/> parameters.</item>
+    /// </list>
+    /// </para>
+    /// <para>
+    /// This attribute is used in those instances where the <see langword="ref"/> should be allowed to escape.
+    /// </para>
+    /// </remarks>
+    [AttributeUsageAttribute(
+        AttributeTargets.Method | AttributeTargets.Property | AttributeTargets.Parameter,
+        AllowMultiple = false,
+        Inherited = false)]
+    public sealed class UnscopedRefAttribute : Attribute
```

Comment:
We can hold off on doing so until it's needed, but we may need to do the same visibility dance we do with other attributes that may get compiled into other projects downlevel, e.g.
https://github.com/dotnet/runtime/blob/cd25e3b7ff768a17b7f062494df239fa12cfa9f4/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/RequiresDynamicCodeAttribute.cs#L14-L18